crates/iroha_executor/src/permission.rs

```diff
@@ -1114,3 +1116,37 @@ pub(crate) fn roles_permissions(host: &Iroha) -> impl Iterator<Item = (RoleId, P
                 .map(move |permission| (role.id().clone(), permission))
         })
 }
+
+/// Revoked all permissions satisfied given [condition].
+///
+/// Note: you must manually check after this function:
+/// `if executor.verdict().is_err() { return; }`
+pub(crate) fn revoke_permissions<V: Execute + ?Sized>(
+    executor: &mut V,
+    condition: impl Fn(&PermissionObject) -> bool,
+) {
```

Comment:
An alternative may be this function returns `Result<(), ValidationFail>`, which would force the caller to handle `Err` case

Reply:
I don't think those revokes are currently expected to be infallible. For example, who `CanUnregisterDomain` can fail to revoke `CanModifyDomainMetadata` in `visit_unregister_domain`.
Or we could consider this as a bug that should be eliminated in another issue, and make it panic or `ValidationFail::InternalError`

Reply:
Yes, it looks odd to me too that someone with such great permission like unregister would not able to revoke the associated permissions

Reply:
let's make this a separate issue and merge

Reply:
#5244